app/gui/src/dashboard/services/RemoteBackend.ts

```diff
@@ -1471,6 +1471,36 @@ export default class RemoteBackend extends Backend {
     }
   }
 
+  /** Download the project to a temporary location. */
+  async downloadProject(id: backend.ProjectId): Promise<DirectoryId> {
+    const details = await this.getProjectDetails(id, true)
+
+    invariant(details.url != null, 'The download URL of the project must be present.')
+
+    const queryString = new URLSearchParams({
+      downloadUrl: details.url,
+    })
+
+    const response = await this.client.get(`./api/cloud/download-project?${queryString}`)
+    const path = await response.text()
+
+    invariant(response.ok, 'The download-project response must have status OK.')
```

Comment:
As I mentioned above, the `invariant` is supposed for `uh-oh` situations. Prefer throwing an error here instead